app/Http/Controllers/Document/StarredDocumentsController.php

```diff
@@ -56,6 +56,25 @@ public function index(Guard $auth, Request $request)
 
         $has_starred = Starred::with('document')->ofUser($user->id)->count() > 0;
         
+        //Number of Items per page 
```

Comment:
As the code is equal for every controller you might think to write it once in the `Controller` base class or in the `searchRequestCreate` method of Search trait that is included anywhere we have to make the number of items per page option available.

Reply:
I was thinking also if make sense to investigate another option which is a controller that only stores in the user profile the new value and then redirects back (with `return redirect()->back()`) or just return an ok status so we can make the call via Javascript and just reload the page the user is seeing.